### PR TITLE
Move remaining reconcilers to new structure.

### PR DESCRIPTION
**Pull request by n3wscott**
Followup to https://github.com/knative/eventing-sources/pull/198

**Release Note**

```release-note
None
```

### CODE REVIEW

**Comment by n3wscott** [comment]
/hold 

until https://github.com/knative/eventing-sources/pull/198 lands

**Comment by n3wscott** [comment]
/retest

tell me what is wrong!

**Comment by n3wscott** [comment]
/assign @grantr 

Take a look!

File: contrib/gcppubsub/pkg/reconciler/gcppubsubsource.go [hunk 1]
```diff
@@ -41,9 +46,41 @@ import (
...
+
+	// raImageEnvVar is the name of the environment variable that contains the receive adapter's
+	// image. It must be defined.
+	raImageEnvVar = "GCPPUBSUB_RA_IMAGE"
+
 	finalizerName = controllerAgentName
 )
 
+// Add creates a new GcpPubSubSource Controller and adds it to the Manager with
+// default RBAC. The Manager will set fields on the Controller and Start it when
```
Comment (vaikas-google) [inline contrib/gcppubsub/pkg/reconciler/gcppubsubsource.go]: What does 'default RBAC' mean?

Comment (vaikas-google) [reply contrib/gcppubsub/pkg/reconciler/gcppubsubsource.go]: what fields will Manager set on the Controller? Maybe a pointer to them?

File: pkg/reconciler/awssqssource/awssqssource.go [hunk 2]
```diff
@@ -38,12 +39,53 @@ import (
...
+
+	// raImageEnvVar is the name of the environment variable that
+	// contains the receive adapter's image. It must be defined.
+	raImageEnvVar = "AWSSQS_RA_IMAGE"
+
 	finalizerName = controllerAgentName
 )
 
+// Add creates a new AwsSqsSource Controller and adds it to the Manager with
+// default RBAC. The Manager will set fields on the Controller and Start it when
```
Comment (vaikas-google) [inline pkg/reconciler/awssqssource/awssqssource.go]: ditto here.

File: pkg/reconciler/kuberneteseventsource/kuberneteseventsource.go [hunk 3]
```diff
@@ -18,21 +18,91 @@ package kuberneteseventsource
...
+func newReconciler(mgr manager.Manager, receiveAdapterImage string) reconcile.Reconciler {
+	return &reconciler{
+		Client:              mgr.GetClient(),
+		scheme:              mgr.GetScheme(),
+		recorder:            mgr.GetRecorder(controllerAgentName),
+		receiveAdapterImage: receiveAdapterImage,
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
```
Comment (vaikas-google) [inline pkg/reconciler/kuberneteseventsource/kuberneteseventsource.go]: Maybe say something here that r will reconcile based on changes to owned ContainerSource as well as KubernetesEventSource.

File: contrib/gcppubsub/pkg/reconciler/gcppubsubsource.go (hunk 1 above)
Comment (n3wscott) [reply contrib/gcppubsub/pkg/reconciler/gcppubsubsource.go]: Add clarity to the comment for gcppubsubsource.Add()

Comment (n3wscott) [reply contrib/gcppubsub/pkg/reconciler/gcppubsubsource.go]: https://github.com/knative/eventing-sources/issues/202

File: pkg/reconciler/awssqssource/awssqssource.go (hunk 2 above)
Comment (n3wscott) [reply pkg/reconciler/awssqssource/awssqssource.go]: Add clarity to the comment for awssqusource.Add()

Comment (n3wscott) [reply pkg/reconciler/awssqssource/awssqssource.go]: https://github.com/knative/eventing-sources/issues/203

File: pkg/reconciler/kuberneteseventsource/kuberneteseventsource.go (hunk 3 above)
Comment (n3wscott) [reply pkg/reconciler/kuberneteseventsource/kuberneteseventsource.go]: Add clarity to the comment for kuberneteseventsource.add()

Comment (n3wscott) [reply pkg/reconciler/kuberneteseventsource/kuberneteseventsource.go]: https://github.com/knative/eventing-sources/issues/204

**Comment by n3wscott** [comment]
wake up tide!

**Comment by bbrowning** [comment]
Kicking tide to see if it's alive. It should complain at me for this:

/approve

**Comment by bbrowning** [comment]
That did something, so kicking it again.

/lgtm

**Comment by n3wscott** [comment]
🎉 !!

**Pull request merged**